docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,698 @@ id: pillars-turbomodules
 title: TurboModules
 ---
 
-This section contains a high-level introduction to TurboModules. It provides enough context to understand when a TurboModule is needed and how it roughly works.
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
 
-This section must have a warning that it works only with the new architecture enabled. It points to the [migration section](../new-architecture-intro).
+If you've worked with React Native, you may be familiar with the concept of [Native Modules](../native-modules-intro.md), which allow JavaScript and platform-native code to communicate over the React Native "bridge", which handles cross-platform serialization via JSON.
 
-## How to create a Turbomodule
+TurboModules are the next iteration on Native Modules that provide a few extra [benefits](./why):
 
-This section is a step-by-step guide to create a TurboModule from scratch. The list of subsections is roughly:
+- Strongly typed interfaces that are consistent across platforms
+- The ability to create shared C++ code for use across platforms
+- Lazy loading of modules, allowing for faster app startup
+- The use of JSI, a JavaScript interface for native code, which allows for more efficient communication between native and JavaScript code than the bridge
 
-- JS spec (with all the supported features)
-- Configuration (package.json, cocoapods, gradle, …) and CodeGen
-- Native code (one section for iOS and one for Android)
-- Integration in an App (`yarn add` and how to connect the JS specs to the app itself)
-- Troubleshooting (common issues and how to solve them)
+This guide will show you how to create a basic TurboModule.
+
+:::caution
+TurboModules only work with the **New Architecture** enabled.
+To migrate to the **New Architecture**, follow the [Migration guide](../new-architecture-intro)
+:::
+
+## How to Create a TurboModule
+
+To create a TurboModule, we need to:
+
+1. Define the JavaScript specification.
+2. Configure the module so that CodeGen can generate the scaffolding.
+3. Write the native code to finish implementing the module.
+
+## 1. Folder Setup
+
+In order to keep the module decoupled from the app, it's a good idea to define the module separately from the app, and then add it as a dependency to your app later. This is also what you'll do for writing TurboModules that can be released as open-source libraries later.
+
+Next to your application, create a folder called `RTNCalculator`. (**RTN** stands for "**R**eact**T** **N**ative", and is a standard prefix for React Native modules).
```

Comment:
I think "standard" is also okay to mean what you're conveying here ("a standard" can mean "a required method of doing thing", but it can also mean "conventional"). But "conventional prefix" is also okay, but it sounds more awkward to me.

Reply:
Addressed, chose "recommended" as I think it conveys the idea better